### PR TITLE
feat: add back rng arg for conditional parameter generation

### PR DESCRIPTION
**Pull request by cryptonemo**

### CODE REVIEW

**Comment by cryptonemo** [comment]
This PR undoes the removal of the conditional parameter generation (originally removed in https://github.com/filecoin-project/rust-fil-proofs/pull/986/)

**Comment by cryptonemo** [comment]
Sorry, I see I missed the --ignored tests.  Will resolve.

**Comment by cryptonemo** [comment]
Hrm, interestingly the failed CI tests pass locally.

**Comment by cryptonemo** [comment]
> Hrm, interestingly the failed CI tests pass locally.

Nevermind, I have params ...

File: storage-proofs/src/compound_proof.rs [hunk 1]
```diff
@@ -268,12 +268,18 @@ pub trait CompoundProof<
 
     fn blank_circuit(public_params: &S::PublicParams) -> C;
 
-    fn groth_params(public_params: &S::PublicParams) -> Result<groth16::MappedParameters<E>> {
-        Self::get_groth_params(Self::blank_circuit(public_params), public_params)
+    fn groth_params<R: RngCore>(
+        rng: Option<&mut R>,
```
Comment (dignifiedquire) [inline storage-proofs/src/compound_proof.rs]: lets add a doc comment clearly stating what happens if you pass an rng, vs not, to all these methods

**Comment by cryptonemo** [comment]
> lgtm, except some additional doc comments

Thanks, will update shortly!

**Comment by cryptonemo** [comment]
Very odd that comment changes resulted in test status differences.

**Comment by cryptonemo** [comment]
Regarding the test failures, I can't reproduce that locally, but I do see this which looks suspect:

```
running 4 tests
test porep::drg::compound::tests::test_drgporep_compound_pedersen ... ignored
test porep::drg::compound::tests::test_drgporep_compound_poseidon ... ignored
proof parents were not those provided in public parameters
proof parents were not those provided in public parameters
proof parents were not those provided in public parameters
invalid test data can't scramble proofs with all same parents.
test porep::drg::vanilla::tests::test_drgporep_verifies_using_challenge ... ok
proof parents were not those provided in public parameters
invalid test data can't scramble proofs with all same parents.
proof parents were not those provided in public parameters
test porep::drg::vanilla::tests::test_drgporep_verifies_parents ... ok

test result: ok. 2 passed; 0 failed; 2 ignored; 0 measured; 167 filtered out
```

By any chance, can anyone else reproduce this?

**Comment by cryptonemo** [comment]
As I suspected, re-running the test allowed it to pass.  Given that the output mentioned above is expected (after looking into) and I cannot find a way to reproduce, I am inclined to chalk this up to a CI environment failure, rather than a real bug.  @dignifiedquire If you agree, I'll merge.  Otherwise, let me know how you think I should proceed here.  Thanks!

**Comment by dignifiedquire** [comment]
lets merge it, and keep an eye on any flakyness in CI in upcoming PRs.

**Pull request merged**